app/src/main/java/au/id/neasbey/biblecast/util/BibleAPIRequest.java

```diff
@@ -0,0 +1,177 @@
+
+    private String requestUrl = "";
+
+    private String apiAuth = "";
+
+    private final List<Spanned> resultList = new LinkedList<>();
+
+    public String createRequestUrl(String apiUrl, String apiAuth, String apiQuery, String bibleVersions) {
+        boolean missing = false;
+
+        if(TextUtils.isEmpty(apiUrl))
+        {
+            Log.e(TAG, "No API URL specified");
+            missing = true;
+        }
+
+        if(TextUtils.isEmpty(apiAuth))
+        {
+            Log.e(TAG, "No API authentication specified");
+            missing = true;
+        } else {
+            this.apiAuth = apiAuth;
+        }
+
+        if (TextUtils.isEmpty(apiQuery)) {
+            Log.e(TAG, "No API query specified");
+            missing = true;
+        }
+
+        if (TextUtils.isEmpty(bibleVersions)) {
+            Log.e(TAG, "No API bible versions specified");
+            missing = true;
+        }
+
+        if(!missing) {
+            try {
+                // Set request address
+                requestUrl += apiUrl;
+
+                // Set request parameters
+                requestUrl += "?query=" + URLEncoder.encode(apiQuery, "UTF-8");
+                requestUrl += "&version=" + URLEncoder.encode(bibleVersions, "UTF-8");
+
+            } catch (UnsupportedEncodingException e) {
+                Log.e(TAG, "URL encoding Error: " + e.getMessage());
+            }
+        }
+
+        return requestUrl;
+    }
+
+    public String performRequest() {
+        Log.d(TAG, "Request: " + requestUrl);
+
+        String resultToDisplay = "";
+
+        if(isRequestValid()) {
+            try {
+                HttpURLConnection urlConnection = createConnection(requestUrl, apiAuth);
+
+                checkServerResponseOk(urlConnection);
+                handleResponse(urlConnection);
+
+            } catch (Exception e) {
+                resultToDisplay = e.getMessage();
+            }
+        } else {
+            resultToDisplay = "Please enter a valid query";
+        }
+
+        return resultToDisplay;
+    }
+
+    public boolean isRequestValid() {
+        return !requestUrl.isEmpty();
+    }
+
+    public List<Spanned> getResultsList() {
+        return resultList;
+    }
+
+    private HttpURLConnection createConnection(String requestUrl, String apiAuth) throws IOException {
+        // Send HTTP Get request
+        URL url = new URL(requestUrl);
+        HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+
+        // Set API authentication
+        urlConnection.addRequestProperty("Authorization", generateHttpAuthentication(apiAuth.getBytes()));
+
+        return urlConnection;
+    }
+
+    private void handleResponse(HttpURLConnection urlConnection) throws Exception {
+        String responseJSON = getServerResponseJSON(urlConnection);
+
+        // Parse server response
+        BibleAPIResponseParser bibleAPIResponseParser = new BibleAPIResponseParser();
```

Comment:
Should this dependency be set in the constructor ... creating instances within a method will make things harder to unit test.